django/db/models/lookups.py

```diff
@@ -163,6 +174,18 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(make_hashable(self.identity))
 
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        c = copy(self)
+        c.lhs = self.lhs.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        c.rhs = self.rhs.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        return c
+
+    def select_format(self, compiler, sql, params):
+        return sql, params
+
+    def get_db_converters(self, connection):
+        return []
+
```

Comment:
Have you tried subclassing `Expression` instead of redefining all of these methods? Looks like a lot the `Lookup` boilerplate could go away with

```python
class Lookup(Expression):
    ...

    def __init__(self, lhs, rhs):
        self.lhs, self.rhs = lhs, rhs
        super().__init__(lhs, rhs)
        ...

    @cached_property
    def output_field(self):
        return BooleanField()

    ...
```

Reply:
I will try this today.